core/src/test/java/org/eclipse/hono/util/TenantObjectTest.java

```diff
@@ -229,11 +229,12 @@ public void testGetTrustAnchorForPublicKeyWithoutValidityPeriod() {
                 .put(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA, new JsonObject()
                         .put(TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, "CN=test")
                         .put(TenantConstants.FIELD_PAYLOAD_PUBLIC_KEY, "noBase64".getBytes()));
-        final TenantObject obj = tenantPayload.mapTo(TenantObject.class);
+
         try {
-            assertThat(obj.getTrustAnchors().size(), is(0));
-        } catch (GeneralSecurityException e) {
-            // should never get here
+            tenantPayload.mapTo(TenantObject.class);
+            fail("should not have been able to create a TenantObject from a malformed trusted CA");
+        } catch (final IllegalArgumentException e) {
+            // expected
```

Comment:
@sophokles73: i will like to get you opinion on this: do you think this will cause a back incompatibility issue if existing content in a device registry database does not store the validity period of public keys? currently, Hono does not require this at all and existing data might not have such validity periods recorded

otherwise, we can enforce this check only when storing the data and not when loading it

Reply:
I would enforce it only when storing the data for now.

Reply:
ok